controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -78,6 +79,7 @@
     private final StreamMetrics streamMetrics;
     private final TransactionMetrics transactionMetrics;
 
+    @VisibleForTesting
```

Comment:
This is not visible for testing as it was public already.

Reply:
I introduced this constructor in #3117 to avoid all the existing test classes instantiating `ControllerService` to be changed just because of the two new fields (`streamMetrics`, `transactionMetrics`), which are in fact not relevant for most test classes. As this constructor is used only in testing environment, I thought that this annotation may make that more explicit. But I can remove it if you believe that it is not worth it.